discord_thread_summarizer_using_openai/discord_utils.bal

```diff
@@ -0,0 +1,55 @@
+import ballerina/time;
+import ballerina/http;
+
+const time:Seconds DISCORD_EPOCH = 1420070400000.00;
+
+http:Client discordClient = check new (DISCORD_API_URL);
+
+function getChannelDetails(int channelId) returns Channel|error {
+    string path = string `/channels/${channelId}`;
+    http:Response response = check discordClient->get(path, headers = {
+        "Authorization": botToken
+    });
+    return (check response.getJsonPayload()).cloneWithType();
+}
+
+function getActiveThreads(string guildId) returns ActiveThreads|error {
+    string path = string `/guilds/${guildId}/threads/active`;
+    http:Response response = check discordClient->get(path, headers = {
+        "Authorization": botToken
+    });
+    return (check response.getJsonPayload()).cloneWithType();
+}
+
+function getMessages(string threadId, string timestamp = "") returns Message[]|error {
+    string path = string `/channels/${threadId}/messages`;
+
+    if (timestamp != "") {
+        path = string `/channels/${threadId}/messages?after=${timestamp}`;
+    }
+    http:Response response = check discordClient->get(path, headers = {
+        "Authorization": botToken
+    });
+    return (check response.getJsonPayload()).cloneWithType();
+}
+
+function getThreadURL(string guildId, string threadId) returns string {
+    return string `https://discord.com/channels/${guildId}/${threadId}`;
+}
+
+function reverseArray(Message[] arr) returns Message[] {
+    int len = arr.length();
+    Message[] result = [];
+    foreach int i in 0 ..< len {
+        result.push(arr[len - i - 1]);
+    }
+    return result;
+}
```

Comment:
Why do you need this? Why can't we use the `reverse` function from the array langlib?

Reply:
Addressed in https://github.com/ballerina-guides/ai-samples/pull/69/commits/1dbf76e067fd08cc0430182aa02b7c98d154f3ac